File: Message Encryption.py
s = input().strip()
c = int(input())
s = list(s)
m = [['0' for j in range(c)] for i in range(len(s) // c)]
r = []
i, j = 0, 0
for k in range(len(s)):
    m[i][j] = s[k]
    if j == c - 1:
        i += 1
        j = 0
    else:
        j += 1
for i in range(len(m)):
    m[i] = list(m[i])
    if (i + 1) % 2 == 0:
        m[i].reverse()
    r.append(m[i])
i, j = 0, 0
for k in range(len(s)):
    print(r[i][j], end="")
    if i == len(s) // c - 1:
        i = 0
        j += 1
        continue
    i += 1

Example 1:
Input:
midinadiazne
3
Output:
madeinindiaz
Example 2:
Input:
loaesfbnaiordilertenrdhdw
5
Output:
lionroaredandthebirdsflew

File: Message Encryption.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
    string s;
    int c;
    getline(cin, s);
    cin >> c;

    int r = s.length() / c;  // Number of rows
    vector<vector<char>> m(r, vector<char>(c));
    int idx = 0;

    // Fill the matrix row by row
    for (int i = 0; i < r; i++) {
        for (int j = 0; j < c; j++) {
            m[i][j] = s[idx++];
        }
    }

    // Reverse the characters in even rows (1-based index)
    for (int i = 1; i < r; i += 2) {
        for (int j = 0, k = c - 1; j < k; j++, k--) {
            char t = m[i][j];
            m[i][j] = m[i][k];
            m[i][k] = t;
        }
    }

    // Collect the result by traversing column-wise
    string res = "";
    for (int j = 0; j < c; j++) {
        for (int i = 0; i < r; i++) {
            res += m[i][j];
        }
    }

    // Print the final result
    cout << res << endl;

    return 0;
}

Example 1:
Input:
midinadiazne
3
Output:
madeinindiaz
Example 2:
Input:
loaesfbnaiordilertenrdhdw
5
Output:
lionroaredandthebirdsflew
